src/odemis/driver/xt_client.py

```diff
@@ -1506,15 +1550,28 @@ def _acquire(self):
                 while True:
                     if self._acq_should_stop():
                         break
-                    self.parent.set_channel_state(self._channel_name, True)
-                    if self.parent._scanner.blanker.value is None:
-                        self.parent.unblank_beam()
+                    self.parent.set_channel_state(self._scanner.channel, True)
+                    if hasattr(self._scanner, "blanker"):
+                        # TODO: When the acquisition ends, if blanker is set to automatic (None), we need to activate
+                        #  the blanker again. That should also happen when switching e-beam <--> FIB, but currently
+                        #  the FIB blanker is not support via the XT interface.
+                        if self._scanner.blanker.value is None:
+                            self.parent.unblank_beam()
                     # The channel needs to be stopped to acquire an image, therefore immediately stop the channel.
-                    self.parent.set_channel_state(self._channel_name, False)
+                    self.parent.set_channel_state(self._scanner.channel, False)
+                    md = self._scanner._metadata.copy()
 
                     # Estimated time for an acquisition is the dwell time times the total amount of pixels in the image.
-                    n_pixels = self.parent._scanner.shape[0] * self.parent._scanner.shape[1]
-                    est_acq_time = self.parent._scanner.dwellTime.value * n_pixels
+                    if hasattr(self._scanner, "dwellTime") and hasattr(self._scanner, "shape"):
```

Comment:
`hasattr(self._scanner, "shape")` is confusing. You're not using `shape`, and anyway, all scanners have a .shape. 
-> `hasattr(self._scanner, "resolution")` ?
Maybe even clearer to move the metadata update in a separate if, just like rotation. Otherwise, in the (theoretical) case that there is dwellTime, but no resolution, the MD_DWELL_TIME will not be set, but it should.